frontend/lib/src/components/core/Block/Block.tsx

```diff
@@ -146,6 +148,8 @@ const BlockNodeRenderer = (props: BlockPropsWithWidth): ReactElement => {
 }
 
 const ChildRenderer = (props: BlockPropsWithWidth): ReactElement => {
+  // Handle cycling of colors for dividers:
+  assignDividerColor(props.node, useTheme())
```

Comment:
not super happy about having header element-specific logic here, but I don't have any better idea 🙈

Reply:
Completely fair, I battled with where to put this as product confirmed the color cycling is important.
Also, this is another instance lately of adding element checking logic (other example `containsChatInput`), perhaps we could discuss where/if to consolidate these things going forward

Reply:
Yep indeed, chat input was also one of the candidates that needed some ugly checks outside its own implementation  😬

Reply:
one quick idea (but not sure if better or feasible): we could theoretically add a mutable property (e.g. `nextDividerColor`) to the [ElementNode](https://github.com/streamlit/streamlit/blob/217893e27ba591015d6e900cf007b5bbcfb6a665/frontend/lib/src/AppNode.ts#L123) and then we move all the color picking logic to the header component instead.

Reply:
E.g. in ElementNodeRenderer:

```
 case "heading":
      return (
        <Heading
          width={width}
          element={node.element.heading as HeadingProto}
          nextDividerColor={node.nextDividerColor} <- This needs to be some kind of mutable object
        />
      )
```

Reply:
And in the `Heading` it changes the `nextDividerColor` to the next color in case it uses an auto color divider

Reply:
Oh wait, that probably won't work since node here is a single element and not the parent block 😬

Reply:
Just found that out the long way haha

Reply:
Ups, sorry for misleading 😔

Reply:
No worries at all! Definitely worth exploring. 
Are you comfortable with merge though?

Reply:
Yep, sure 👍 let's merge it in the current state